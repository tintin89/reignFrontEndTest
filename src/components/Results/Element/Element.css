.element{
    cursor: pointer;
    margin: 30px 40px;
    width: 550px;
    height: 90px;
    display: flex;  
    align-items: center;
    opacity: 0.8;
    border-radius: 6px;
    border: solid 1px #979797;
    background-color: #fff;
    transition: ease-out 0.3s;
}
.element:hover{
    opacity: 1;
}
.info{
    padding-left: 25px;    
    flex: 1;
    display: flex;
    flex-flow: column;
}

.timeInfo{
    display: flex;
    align-items: center;
}
.timeInfo span{
    margin-left: 0.5rem;    
    font-family: Roboto;
    font-size: 11px;
    font-weight: normal;
    font-stretch: normal;
    font-style: normal;
    line-height: normal;
    letter-spacing: normal;
    color: #767676;
}

.info h3{
  margin-top: 0.5rem;
  font-family: Roboto;
  font-size: 14px;
  font-weight: 500;
  font-stretch: normal;
  font-style: normal;
  line-height: 1.43;
  letter-spacing: 0.25px;
  color: #6b6b6b;
}

.favs{    
    flex: 0.2;
    height: 100%;
    background-color: #60606011;
    display: flex;
    align-items: center;
    justify-content: center;
 
}


.favs img{
    width: 24px;
    height: 24px;
   position: absolute;
}