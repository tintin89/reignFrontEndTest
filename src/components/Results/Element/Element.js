import React,{useState} from 'react';
import './Element.css';
import favEmpty from '../../../assets/iconmonstr-favorite-2.svg';
import favfull  from '../../../assets/iconmonstr-favorite-3.svg';
import clock from '../../../assets/iconmonstr-time-2.svg';
import { isFavorite } from '../../../utility';
import {useDispatch,useSelector} from 'react-redux';
import {updateFavorites} from '../../../store/actions';
import moment from 'moment';

const mapState = (myStateApp)=>({
    favorites:myStateApp.favorites
})

function Element({title,id,created,author}) {   
    const {favorites} = useSelector(mapState);
    const dispatch = useDispatch();
    const [fav,setFav] = useState(localStorage.getItem("favorites") ? isFavorite(id) : false);

    const handleFavorite = ()=>{        
        const favTemp = favorites;
        if(!fav){
            setFav(true);            
            favTemp.push({title,id,created,author})
            localStorage.setItem("favorites",JSON.stringify(favTemp));
            dispatch(updateFavorites(favTemp));
        }else{
            setFav(false);
            localStorage.setItem("favorites",JSON.stringify(favTemp.filter(e=>e.id!==id)));
            const filterFav = favTemp.filter(e=>e.id!==id);
            dispatch(updateFavorites(filterFav));         
        }
        
        
    }
    
    
    return (
        <div  className="element">
           <div className="info">
                <div className="timeInfo">
                 <img alt="clock" src={clock}/>
                 <span>{created+" by "+author}</span>
                </div>
               <h3>{title}</h3>
           </div>
           <div onClick={()=>handleFavorite()} className="favs">
           <img alt="title"  src={fav ? favfull : favEmpty}/>
           </div>
        </div>
    )
}

export default Element
