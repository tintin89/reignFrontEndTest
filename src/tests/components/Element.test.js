import React from 'react';
import  { mount}  from "enzyme";
import Element from '../../components/Results/Element/Element';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';

const mockStore = configureStore();

describe('tests in  <Dropdown/>',()=>{   
    const initState = {
        favorites:[],        
    };
    let store = mockStore(initState);

    test('Must show correctly the component',()=>{
       
        const wrapper = mount(
            <Provider store={store}>
             <Element/>
            </Provider>
        );
        expect(wrapper).toMatchSnapshot()
    }) 
   
})