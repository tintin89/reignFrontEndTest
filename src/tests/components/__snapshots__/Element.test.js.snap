// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests in  <Dropdown/> Must show correctly the component 1`] = `
<div
  className="element"
>
  <div
    className="info"
    onClick={[Function]}
  >
    <div
      className="timeInfo"
    >
      <img
        alt="clock"
        src="iconmonstr-time-2.svg"
      />
      <span>
        undefined by undefined
      </span>
    </div>
    <h3 />
  </div>
  <div
    className="favs"
    onClick={[Function]}
  >
    <img
      alt="title"
      src="iconmonstr-favorite-2.svg"
    />
  </div>
</div>
`;
